app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,18 @@
+class CategoriesController < ApplicationController
+	def index
+		@categories_all = Category.all
+		# users = User.where(id: current_user.id)
+		# lessons = Lesson.where(user_id: current_user.id)
+		# num = lessons.pluck(:category_id)
+		@categories_done = current_user.categories
+
+		@categories = @categories_all - @categories_done
+	end
+
+	def show
+		@category = Category.find(params[:id])
+		lessons = @category.lessons
+		@lesson = lessons.find_by(user_id: current_user.id)
```

Comment:
@YuheiTsunoda You can improve this code to one line:
`@lesson = current_user.lessons.find_by(category_id: params[:id])`